File: leetcode-75/day-1/1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.js
/**
 * Runtime: 59ms
 * Memory Usage: 31.3 MB
 * Link: https://leetcode.com/submissions/detail/756400353/
 * //-----------------------------------------------------------------------------
 *    1. Loop over the nums
 *    2.    Add to nums[i] previous element
 *    3. return nums
 * //-----------------------------------------------------------------------------
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {

    // 1. Loop over the nums
    for(let i = 1; i < nums.length; i++) {

        // 2. Add to nums[i] previous element
        nums[i]  += nums[i - 1];
    }

    // 3. return nums
    return nums;
};

File: leetcode-75/day-1/1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.cs
/**
 * Runtime: 148 ms
 * Memory Usage: 43.4 MB
 * Link: https://leetcode.com/submissions/detail/756419665/
//-----------------------------------------------------------------------------
   1. Loop over the nums
   2.    Add to nums[i] previous element
   3. return nums
//-----------------------------------------------------------------------------
 * @param {number[]} nums
 * @return {number[]}
 */
public class Solution {
    public int[] RunningSum(int[] nums) {

        // 1. Loop over the nums
        for(int i = 1; i < nums.Length; i++)
        {
            // 2. Add to nums[i] previous element
            nums[i] += nums[i - 1];
        }

        // 3. return nums
        return nums;
    }
}